packages/nimble-components/src/list-option-group/tests/list-option-group.spec.ts

```diff
@@ -0,0 +1,463 @@
+        element.appendChild(slottedLabel);
+        await waitForUpdatesAsync();
+
+        expect(element.labelContent).toBe('Group 1');
+
+        await disconnect();
+    });
+
+    it('if multiple slotted labels are provided with extra whitespace, labelContent result is trimmed', async () => {
+        const { element, connect, disconnect } = await setup();
+        await connect();
+        element.label = undefined;
+        const slottedLabel1 = document.createElement('span');
+        slottedLabel1.textContent = '  Slotted Label 1 ';
+        element.appendChild(slottedLabel1);
+        const slottedLabel2 = document.createElement('span');
+        slottedLabel2.textContent = '     Slotted Label 2 ';
+        element.appendChild(slottedLabel2);
+        await waitForUpdatesAsync();
+
+        expect(element.labelContent).toBe('Slotted Label 1 Slotted Label 2');
+        await disconnect();
+    });
+
+    it('group is visually hidden with no options', async () => {
+        const { element, connect, disconnect } = await setup();
+        await connect();
+        expect(element.visuallyHidden).toBeTrue();
+        await disconnect();
+    });
+
+    it('group is visually hidden with no visible options', async () => {
+        const { element, connect, disconnect } = await setup();
+        await connect();
+        const option = document.createElement('nimble-list-option');
+        option.hidden = true;
+        element.appendChild(option);
+        const option2 = document.createElement('nimble-list-option');
+        option2.visuallyHidden = true;
+        element.appendChild(option2);
+        await waitForUpdatesAsync();
+        expect(element.visuallyHidden).toBeTrue();
+        await disconnect();
+    });
+
+    it('group is visible with visible options', async () => {
+        const { element, connect, disconnect } = await setup();
+        await connect();
+        const option = document.createElement('nimble-list-option');
+        element.appendChild(option);
+        await waitForUpdatesAsync();
+        expect(element.visuallyHidden).toBeFalse();
+        await disconnect();
+    });
+
+    it('group becomes hidden when all options are removed from it', async () => {
+        const { element, connect, disconnect } = await setup();
+        await connect();
+        const option = document.createElement('nimble-list-option');
+        element.appendChild(option);
+        await waitForUpdatesAsync();
+        expect(element.visuallyHidden).toBeFalse();
+
+        element.removeChild(option);
+        await waitForUpdatesAsync();
+        expect(element.visuallyHidden).toBeTrue();
+        await disconnect();
+    });
+
+    it('option that is removed and then updates its hidden state, does not update group', async () => {
+        const { element, connect, disconnect } = await setup();
+        await connect();
+        const option1 = document.createElement('nimble-list-option');
```

Comment:
we aren't super consistent but can you do a pass and try and use `listOptionTag` for these?